forest/benchmarking/direct_fidelity_estimation.py

```diff
@@ -12,31 +12,78 @@
 from forest.benchmarking.utils import prepare_prod_pauli_eigenstate
 from pyquil import Program
 from pyquil.api import BenchmarkConnection, QuantumComputer
-from pyquil.operator_estimation import ExperimentSetting, TomographyExperiment, \
-    TensorProductState, plusX, minusX, plusY, minusY, plusZ, minusZ
+from pyquil.operator_estimation import ExperimentResult, ExperimentSetting, TomographyExperiment, \
+    TensorProductState, measure_observables, plusX, minusX, plusY, minusY, plusZ, minusZ
 from pyquil.paulis import PauliTerm, PauliSum, sI, sX, sY, sZ
 
 
-def calibrate_readout_imperfections(pauli: PauliTerm, quantum_machine: QuantumComputer,
-                                    var: float = 0.01):
-    """
-    Compute the expectation value and variance of measuring a pauli without any pre or post circuit.
-    This can be used to attribute lack of of visibility to readout imperfections.
-
-    TODO:
-      + look into whether we can calibrate multiple PauliTerm with one call
-      + test how well this calibration works with correlated readout errors
-
-    :param pauli: The Pauli expectation to calibrate.
-    :param quantum_machine: The QVM or QPU to use for calibration.
-    :param var: The desired variance of the result.
-    :return: The expectation, variance, and number of shots of the expectation value of a +1
-     eigenstate of pauli.
-    """
-    this_prog = prepare_prod_pauli_eigenstate(pauli)
-    expectation, variance, count = est.estimate_locally_commuting_operator(
-        this_prog, PauliSum([pauli]), var, quantum_machine)
-    return expectation[0], variance[0, 0].real, count
+@dataclass 
+class DFEexperiment:
+    """Experiments to obtain an estimate of fidelity via DFE"""
+
+    exp: TomographyExperiment
+    """The experiment to be performed"""
+
+    type: str
```

Comment:
Is it a problem that "type" is a reserved word?

Also "results" and "experiments" are abbreviated but e.g. "dimension" is not.

Reply:
Fixed